apps/client/src/app/util/validation.util.ts

```diff
@@ -0,0 +1,27 @@
+import { FormGroup } from '@angular/forms';
+import { plainToInstance } from 'class-transformer';
+import { validate } from 'class-validator';
+
+export function validateObjectForForm<T>(
+  obj: T,
+  ctor: { new (): T },
+  form: FormGroup,
+  onSuccess: () => void
+): void {
+  const objInstance = plainToInstance(ctor, obj);
+
+  validate(objInstance as object).then((errors) => {
+    if (errors.length === 0) {
+      onSuccess();
```

Comment:
I'm not a fan of passing a callback. Is it possible to change the function to return a promise and `resolve()` here?

Reply:
Changed to return a promise that will resolve if there are no errors, otherwise reject